### PR TITLE
Add /ids, /count and /reload endpoints to app

### PR DESCRIPTION
**Pull request by scott-ace-newton**

### CODE REVIEW

File: handlers.go [hunk 1]
```diff
@@ -78,3 +79,41 @@ func (h *genresHandler) GoodToGo(writer http.ResponseWriter, req *http.Request)
 		writer.WriteHeader(http.StatusServiceUnavailable)
 	}
 }
+
+func (h *genresHandler) getCount(writer http.ResponseWriter, req *http.Request) {
+	count := h.service.getLocationCount()
```
Comment (nwrigh) [inline handlers.go]: "count := h.service.getLocationCount()"

File: handlers.go [hunk 2]
```diff
@@ -78,3 +79,41 @@ func (h *genresHandler) GoodToGo(writer http.ResponseWriter, req *http.Request)
...
+	_, err := writer.Write([]byte(strconv.Itoa(count)))
+	if err != nil {
+		log.Warnf("Couldn't write count to HTTP response. count=%d %v\n", count, err)
+		writer.WriteHeader(http.StatusInternalServerError)
+	}
+}
+
+func (h *genresHandler) getIds(writer http.ResponseWriter, req *http.Request) {
+	ids := h.service.getLocationIds()
```
Comment (nwrigh) [inline handlers.go]: Location again

File: handlers.go [hunk 3]
```diff
@@ -78,3 +79,41 @@ func (h *genresHandler) GoodToGo(writer http.ResponseWriter, req *http.Request)
...
+	if len(ids) == 0 {
+		writer.WriteHeader(http.StatusOK)
+		return
+	}
+	enc := json.NewEncoder(writer)
+	type locationId struct {
+		ID string `json:"id"`
+	}
+	for _, id := range ids {
+		rID := locationId{ID: id}
```
Comment (nwrigh) [inline handlers.go]: locationId

File: service.go [hunk 4]
```diff
@@ -14,6 +14,9 @@ type genreService interface {
 	getGenres() ([]genreLink, bool)
 	getGenreByUUID(uuid string) (genre, bool)
 	checkConnectivity() error
+	getLocationCount() int
```
Comment (nwrigh) [inline service.go]: Location again

File: service.go [hunk 5]
```diff
@@ -14,6 +14,9 @@ type genreService interface {
...
+	getLocationIds() []string
```
Comment (nwrigh) [inline service.go]: Location

File: service.go [hunk 6]
```diff
@@ -85,3 +88,40 @@ func (s *genreServiceImpl) initGenresMap(terms []interface{}) {
 		s.genreLinks = append(s.genreLinks, genreLink{APIURL: s.baseURL + top.UUID})
 	}
 }
+
+func (s *genreServiceImpl) getLocationCount() int {
```
Comment (nwrigh) [inline service.go]: Location

File: service.go [hunk 7]
```diff
@@ -85,3 +88,40 @@ func (s *genreServiceImpl) initGenresMap(terms []interface{}) {
...
+	return len(s.genreLinks)
+}
+
+func (s *genreServiceImpl) getLocationIds() []string {
```
Comment (nwrigh) [inline service.go]: Location

File: service.go [hunk 8]
```diff
@@ -85,3 +88,40 @@ func (s *genreServiceImpl) initGenresMap(terms []interface{}) {
...
+		keys[i] = k
+		i++
+	}
+	return keys
+}
+
+func (s *genreServiceImpl) reload() error {
+	s.genresMap = make(map[string]genre)
+	responseCount := 0
+	log.Println("Fetching locations from TME")
```
Comment (nwrigh) [inline service.go]: Location

File: service.go [hunk 9]
```diff
@@ -85,3 +88,40 @@ func (s *genreServiceImpl) initGenresMap(terms []interface{}) {
...
+		}
+
+		if len(terms) < 1 {
+			log.Println("Finished fetching locations from TME")
+			break
+		}
+		s.initGenresMap(terms)
+		responseCount += s.maxTmeRecords
+	}
+	log.Printf("Added %d location links\n", len(s.genreLinks))
```
Comment (nwrigh) [inline service.go]: Location

**Review by carlosroman — APPROVED** [review]

**Review by tosan88 — APPROVED** [review]

**Review by nwrigh — CHANGES_REQUESTED** [review]

**Review by nwrigh — APPROVED** [review]

**Pull request merged**